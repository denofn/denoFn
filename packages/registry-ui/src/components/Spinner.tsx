import React from "react";
import { css, keyframes } from "otion";

type SpinnerProps = {
  dim?: string;
};

const spin = keyframes({
  from: {
    transform: "rotate(0deg)",
  },
  to: {
    transform: "rotate(360deg)",
  },
});

export function Spinner({ dim }: SpinnerProps) {
  return (
    <div
      className={css({
        display: "inline-block",
        transitionTimingFunction: "linear",
        width: dim || "1.5rem",
        height: dim || "1.5rem",
        borderWidth: ".25rem",
        borderColor: "#edf2f7",
        borderStyle: "solid",
        borderTopWidth: ".25rem",
        borderTopColor: "black",
        borderRadius: "9999px",
        animation: `${spin} 1.5s linear infinite`,
      })}
    />
  );
}
