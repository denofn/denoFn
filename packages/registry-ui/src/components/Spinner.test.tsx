import { render, cleanup } from "@testing-library/react";
import React from "react";

import { Spinner } from "./Spinner";

afterEach(cleanup);

it("renders with the correct styles", () => {
  const { asFragment } = render(<Spinner />);
  expect(asFragment()).toMatchSnapshot();
});
