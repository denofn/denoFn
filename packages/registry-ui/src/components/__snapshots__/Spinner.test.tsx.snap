// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the correct styles 1`] = `
<DocumentFragment>
  <div
    class="_1r5gb7q _14ztwpi _pmoan0 _eqx0xi _ye6num _11x1ul3 _fxc8cu _1i432rq _1acr3ak _j033lg _13nzo53"
  />
</DocumentFragment>
`;
